internal/agent/agent.go

```diff
@@ -92,6 +92,9 @@ func (a *Agent) Run(ctx context.Context) error {
 	}
 	healthChecker.Start(healthCheckCh)
 
+	collector := NewCollector(a.log, a.config.DataDir)
+	go collector.collect()
```

Comment:
the context needs to be passed into the collector to be stopped.

Reply:
Why do you think? Currently we keep running the agent, which monitor the changes of inventory.json. Maybe we should keep it running just with noop, because noone can update the inventory.json, but if we keep it running podman-auto-update can update image and restart the pod, so we actually do update the invetory.json.

Reply:
I was speaking about stopping the goroutine. If you don't check the context it will leak at the end. Also, we must wait for the collector to close the inventory.json properly before exiting the agent.

Reply:
Ah, ok. Done.